common/pkg/capi/clustertopology/handlers/cluster_context.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package handlers
+
+import (
+	"context"
+	"errors"
+
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// clusterKeyContextKey is how we find a cluster key from a context.Context.
+type clusterKeyContextKey struct{}
```

Comment:
Just curious why a struct instead of a string constant?

Reply:
To prevent conflicts. As this is just an arbitrary string, other code could inject stuff into this key. Using an unexported struct like this makes that safer.